cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -362,50 +367,140 @@ __global__ void computeSplitClassificationKernel(
     col = select(colIndex, treeid, node.info.unique_id, seed, input.N);
   }
 
-  for (IdxT i = threadIdx.x; i < len; i += blockDim.x) shist[i] = 0;
+  // populating shared memory with initial values
+  for (IdxT i = threadIdx.x; i < pdf_shist_len; i += blockDim.x)
+    pdf_shist[i] = 0;
+  for (IdxT j = threadIdx.x; j < cdf_shist_len; j += blockDim.x)
+    cdf_shist[j] = 0;
+  for (IdxT i = threadIdx.x; i < pdf_shist_len; i += blockDim.x)
+    pdf_shist[i] = 0;
+  for (IdxT j = threadIdx.x; j < cdf_shist_len; j += blockDim.x)
+    cdf_shist[j] = 0;
```

Comment:
there seem to be duplicate initializations?

Reply:
ah some sloppy conflict resolution somewhere, fixed it 👍🏾